robot/impl/resource_manager.go

```diff
@@ -472,6 +472,14 @@ func (manager *resourceManager) completeConfig(
 		if !ok || !gNode.NeedsReconfigure() {
 			continue
 		}
+		if !gNode.ShouldTryReconfigure() {
+			err := errors.Errorf("Reconfiguration error: reached max reconfigure attempts for %s", gNode.Config().Name)
```

Comment:
should we add something to these errors? like maybe saying that we will try again when the config changes